### ISSUE TITLE
when reading FITS table into AstroCatalog - JD is not populated (from header data)

### ISSUE THREAD

**Issue by EranOfek** [EranOfek/AstroPack#555]
when reading FITS table into AstroCatalog - JD is not populated (from header data)

**Comment by agioffe** [comment]
This can be done via reading it again into an AstroHeader (the way I am doing it in some cases), but I would not code it as a default way of reading AstroCatalogs, for in some cases we may not need it at all, and in some other cases we may need some more keywords from the header, not just the JD. Again, we can have it as a non-default option:
if Args.ReadJD
   AH = AstroHeader(FileName,3);
   Obj.HeaderData.replaceVal('JD',AH.getVal('JD'));
end

**Comment by EranOfek** [comment]
this is not that simple.
This is done in AstroTable (line 175) calling ImageIO which already get the
header, but we need another header...

**Comment by agioffe** [comment]
Exactly, that's why in the code above I suggest to read the 3rd header with AH = AstroHeader(FileName,3);

**Comment by EranOfek** [comment]
In a future major release, we may want to allow AstroCatalog object to contain a header.
